test/integration/model/sync.test.js

```diff
@@ -1,408 +1,436 @@
-        name: DataTypes.STRING,
-      });
-      await this.sequelize.sync();
+    await sequelize.sync();
 
-      await this.sequelize.define('testSync', {
-        name: DataTypes.STRING,
-        age: DataTypes.INTEGER,
-        height: { type: DataTypes.INTEGER, field: 'height_cm' },
-      });
+    const descr1 = await testSync.describe();
+    expect(descr1).to.have.ownProperty('name');
 
-      await this.sequelize.sync({ alter: true });
-      const data = await testSync.describe();
-      expect(data).to.have.ownProperty('age');
-      expect(data).to.have.ownProperty('height_cm');
-      expect(data).not.to.have.ownProperty('height');
+    await sequelize.define('testSync', {
+      name: DataTypes.STRING,
+      age: DataTypes.INTEGER,
+      height: { type: DataTypes.INTEGER, field: 'height_cm' },
+    });
+
+    await sequelize.sync({ alter: true });
+    const descr2 = await testSync.describe();
+    expect(descr2).to.have.ownProperty('name');
+    expect(descr2).to.have.ownProperty('age');
+    expect(descr2).to.have.ownProperty('height_cm');
+    expect(descr2).not.to.have.ownProperty('height');
+  });
+
+  it('does not remove columns if drop is set to false in alter configuration', async () => {
+    const testSync = sequelize.define('testSync', {
+      name: DataTypes.STRING,
+      age: DataTypes.INTEGER,
+    });
+    await sequelize.sync();
+
+    await sequelize.define('testSync', {
+      name: DataTypes.STRING,
+    });
+
+    await sequelize.sync({ alter: { drop: false } });
+    const data = await testSync.describe();
+    expect(data).to.have.ownProperty('name');
+    expect(data).to.have.ownProperty('age');
+  });
+
+  it('removes columns if drop is set to true in alter configuration', async () => {
+    const testSync = sequelize.define('testSync', {
+      name: DataTypes.STRING,
+      age: DataTypes.INTEGER,
     });
+    await sequelize.sync();
 
-    it('should not remove columns if drop is set to false in alter configuration', async function () {
-      const testSync = this.sequelize.define('testSync', {
+    await sequelize.define('testSync', {
+      name: DataTypes.STRING,
+    });
+
+    await sequelize.sync({ alter: { drop: true } });
+    const data = await testSync.describe();
+    expect(data).to.have.ownProperty('name');
+    expect(data).not.to.have.ownProperty('age');
+  });
+
+  it('alters a column using the correct column name (#9515)', async () => {
+    const testSync = sequelize.define('testSync', {
+      name: DataTypes.STRING,
+    });
+
+    await sequelize.sync();
+
+    await sequelize.define('testSync', {
+      name: DataTypes.STRING,
+      badgeNumber: { type: DataTypes.INTEGER, field: 'badge_number' },
+    });
+
+    await sequelize.sync({ alter: true });
+    const data = await testSync.describe();
+    expect(data).to.have.ownProperty('badge_number');
+    expect(data).not.to.have.ownProperty('badgeNumber');
+  });
+
+  // IBM i can't alter INTEGER -> STRING
```

Comment:
Does ibmi have something similar that we could test on?

Reply:
I don't know :/ Maybe it can alter between two different string types but can't check without a proper test DB
Note that this wasn't changed by this PR, it's been this way since the initial ibmi PR: https://github.com/sequelize/sequelize/pull/14619/files/77c2f7d81d97413f5a8c7e2ff7d43b2fd2b44ade#diff-81b952d2bbfce09ba3fc0712ea8248f597d3d3421d819a1c69eb06f68fcf318eL108

Reply:
Ah woops, missed that part of the diff. Then we'll hopefully get another chance to look into it once we find somebody that can test on this